zcash_client_sqlite/src/wallet/init/migrations/v_sapling_shard_unscanned_ranges.rs

```diff
@@ -52,17 +53,19 @@ impl<P: consensus::Parameters> RusqliteMigration for Migration<P> {
                 FROM sapling_tree_shards shard
                 LEFT OUTER JOIN sapling_tree_shards prev_shard
                     ON shard.shard_index = prev_shard.shard_index + 1
-                INNER JOIN scan_queue ON 
+                INNER JOIN scan_queue ON
                     (scan_queue.block_range_start BETWEEN subtree_start_height AND shard.subtree_end_height) OR
                     ((scan_queue.block_range_end - 1) BETWEEN subtree_start_height AND shard.subtree_end_height) OR
                     (
                         scan_queue.block_range_start <= prev_shard.subtree_end_height
                         AND (scan_queue.block_range_end - 1) >= shard.subtree_end_height
                     )
-                WHERE scan_queue.priority > {}",
+                INNER JOIN wallet_birthday
+                WHERE scan_queue.priority > {}
+                AND scan_queue.block_range_end > wallet_birthday.height;",
```

Comment:
No, this is correct as it stands. Consider if we modified your suggestion to account for underflow:
```
AND scan_queue.block_range_end >= wallet_birthday.height + 1
```
This is clearly incorrect, because this would mean that a range that has scan_queue.block_range_end == wallet.birthday_height + 1 would be selected, meaning that the birthday height would be scanned.